DDD Workshop.ipynb

```diff
@@ -0,0 +1,398 @@
+    "    airport = fields[0]\n",
+    "    if airport == \"AIRPORT\":\n",
+    "        continue\n",
+    "    time = fields[3]\n",
+    "    \n",
+    "    key = \"{}_{}\".format(airport, time)\n",
+    "    if key not in data_by_airport_time.keys():\n",
+    "        data_by_airport_time[key] = []\n",
+    "    \n",
+    "    occupancy_percent = int(fields[2]) / int(fields[1])\n",
+    "    data_by_airport_time[key].append(occupancy_percent)\n",
+    "    \n",
+    "cleaned_data_file.close()\n",
+    "\n",
+    "rate_by_airport_time = {}\n",
+    "for key, data_arr in data_by_airport_time.items():\n",
+    "    occupancy_average = sum(data_arr) / len(data_arr)\n",
+    "    rate_by_airport_time[key] = occupancy_average\n",
+    "\n",
+    "[print(k, round(v, 4)) for k,v in rate_by_airport_time.items()]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There we go! We can now make a conclusion about which flights would be best to add to the current schedule. From the looks of it, another redeye to JFK would be the best bet here, but this relies on the assumption that the additional supply would produce additional demand. If the demand is static for redeye flights, meaning the supply changing has little effect on the demand, then adding flight here would do very little. It is also possible that the current flights, even though they are the most full, do not have any additional demand for additional seats. These aren't questions we can answer with this dataset alone but might be one you could get from somebody else working at the airport.\n",
+    "\n",
+    "\n",
+    "#### Robot conclusions\n",
+    "\n",
+    "Finally, we can acknowledge that machine learning and artificial intelligence exist. Sometimes, the data-driven decisions we want to be made are not those we make ourselves, but those that we train our robot friends to make for us. There are many considerations for training a machine learning model or informing an AI's guiding parameters, and they are, unfortunately, mostly out of the scope of this piece. It is noteworthy, however, that the same issues that plague human decision-makers are prevalent with machines too. There are also additional artifacts of the mathematical tools that built these machines causing additional strife, such as local minima for gradient descent and overfitting. These things must all be kept in mind when deciding where to feed the output of your ML model.\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Part 3: Takeaways\n",
+    "\n",
+    "## Data Visualizations\n",
+    "\n",
+    "There is a huge elephant in the room here: sometimes data needs to be communicated from the analyst to other people. The analyst might be the one making decisions and might not be, and the decision-maker might be data conscious and might not be, so how do we bridge that gap? An extensive blog post about making data-driven decisions? NO! Data visualizations of course!\n",
+    "\n",
```

Comment:
Is it possible to add a brief data visualization example using flight data here? For example, take the output text data from the previous example and convert it to a bar graph grouped by airport.